### PR TITLE
186114231 performance improvement partial rollback

### PR DESCRIPTION
**Pull request by ttoomey**
## Description

After slack convo, we decided the filter changes needed some additional QA. This PR rolls back the filter changes but leaves in place the migrations and test from 143902a0c30ec409759d3cececfaf576158ac2c5

This also includes a fix for an issue in the filters. Added method to catch and correct this case. Also includes tests against this issue in the household search
 
## Type of change
- [x] Tests improvement
- [x] Bugfix

 
## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by ttoomey** [comment]
@gigxz this is up on staging for QA

File: drivers/hmis/app/models/hmis/filter/base_filter.rb [hunk 1]
```diff
@@ -10,15 +10,16 @@ def initialize(input)
     self.input = input
   end
 
-  def filter_scope(scope)
-    scope
-  end
+  # implement in subclass
+  # def filter_scope(scope)
```
Comment (ttoomey) [inline drivers/hmis/app/models/hmis/filter/base_filter.rb:14]: I don't think we need to define this on the base class(?)

**Review by gigxz — APPROVED** [review]

**Comment by gigxz** [comment]
QA all looks good! Can I merge @ttoomey?

**Comment by ttoomey** [comment]
> QA all looks good! Can I merge @ttoomey?

@gigxz 
Great! Yes please merge, thank you!

**Pull request merged**